chain/chain/src/chain.rs

```diff
@@ -3091,39 +3124,31 @@ impl Chain {
         Ok(())
     }
 
+    /// Collect all the execution outcomes existing at the current moment
+    fn get_partial_recursive_transaction_results(
```

Comment:
To avoid duplication, I would give `get_recursive_transaction_results` parameter like `fail_on_missing_outcome` (could it be shorter?), rename original function to `get_all_recursive_transaction_results` and just call `get_recursive_transaction_results(..., true)` there. `get_partial_recursive_transaction_results` would call `get_recursive_transaction_results(..., false)`.